### PR TITLE
Ruby - ms_rest should throw validation error instead of NameError

### PR DESCRIPTION
**Pull request by vishrutshah**
ms_rest should throw validation error instead of NameError when property not found on ruby object.

Reference: http://apidock.com/ruby/Object/instance_variable_get

### CODE REVIEW

File: src/client/Ruby/ms-rest/lib/ms_rest/serialization.rb [hunk 1]
```diff
@@ -315,7 +315,12 @@ def serialize_composite_type(mapper, object, object_name)
 
         unless model_props.nil?
           model_props.each do |key, value|
-            instance_variable = object.instance_variable_get("@#{key}")
+            begin
+              instance_variable = object.instance_variable_get("@#{key}")
+            rescue NameError
+              fail ValidationError, "instance variable '#{key}' is expected on '#{object}'."
```
Comment (veronicagg) [inline src/client/Ruby/ms-rest/lib/ms_rest/serialization.rb]: should we print object.class?

**Comment by veronicagg** [comment]
@vishrutshah left one comment, rest looks good to me. Thanks!

File: src/client/Ruby/ms-rest/lib/ms_rest/serialization.rb (hunk 1 above)
Comment (vishrutshah) [inline src/client/Ruby/ms-rest/lib/ms_rest/serialization.rb]: Yup that's right. Corrected.

**Comment by veronicagg** [comment]
@vishrutshah LGTM!

**Comment by vishrutshah** [comment]
Thanks for the review!!

**Pull request merged**